Sources/CredentialsJWT/CredentialsJWT.swift

```diff
@@ -91,15 +114,68 @@ public class CredentialsJWT<C: Claims>: CredentialsPluginProtocol {
                         onSuccess(cached.userProfile)
                         return
                     }
+                }
+                
+                do {
+                    _ = try JWT<C>(jwtString: token, verifier: verifier)
+                    
+                    let components = token.components(separatedBy: ".")
+                    guard components.count == 2 || components.count == 3,
+                        let claimsData = Data(base64urlEncoded: components[1]),
+                        let dictionary = try? JSONSerialization.jsonObject(with: claimsData, options: []) as? [String:Any]
```

Comment:
Re. the Swift 4 compilation failure, this is because Swift 5 changed the way optional chaining works (see: https://www.hackingwithswift.com/articles/147/swift-5-is-changing-optional-try)
I'm not sure if we can use a syntax that is going to work in both cases here without resorting to `#if swift(>=5.0)` but maybe try:
```swift
let optionalDict = try? JSONSerialization.jsonObject(with: claimsData, options: []),
let dictionary = optionalDict as? [String:Any]
```